src/relay/pass/type_infer.cc

```diff
@@ -429,6 +522,8 @@ class TypeInferencer::Resolver : public ExprMutator {
     auto it = tmap_.find(GetRef<Expr>(op));
     CHECK(it != tmap_.end());
     Type checked_type = solver_->Resolve(it->second.checked_type);
+    Generalizer gen;
```

Comment:
It seems to be inefficient to create a generalizer for each resolve(note due to the creation of Map) in each pass.